File: src/gestorAplicación/modelos/Proveedor.java
package gestorAplicacion.modelos;

import gestorAplicacion.operaciones.Compra;
import gestorAplicacion.operaciones.Venta;

import java.io.Serializable;
import java.lang.reflect.Array;
import java.util.ArrayList;

public class Proveedor extends Persona implements Serializable {
    private static final long serialVersionUID = 5L;

    private ArrayList<Compra> historico = new ArrayList<Compra>();


    public Proveedor(String nombre, String documento, String email) {
        super(nombre, documento, email);
    }


    public void agregarCompra(Compra compra) {
        historico.add(compra);
    }


    public double calcularFidelidad(int numeroVentasTienda) {
        return historico.size() / numeroVentasTienda;
    }


}

File: python/gestorAplicacion/modelos/Proveedor.py
from gestorAplicacion.modelos.TipoProducto import TipoProducto
from gestorAplicacion.modelos.Persona import Persona


class Proveedor(Persona):
    def __init__(self, nombre, documento, email, productoVente):
        super().__init__(nombre, documento, email)
        self._productoVende = productoVente
        self._historicoProveedor = []

    def getProductoVende(self):
        return self._productoVende

    def getHistoricoProveedor(self):
        return self._historicoProveedor

    def setHistoricoProveedor(self, historicoProveedor):
        self._historicoProveedor = historicoProveedor
